packages/react-data-grid/src/stateManagement/Connect.js

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import EventDispatcher from './EventDispatcher';
+
+const events = new EventDispatcher();
+
+const sharedState = {
+  selectedPosition: {
+    idx: -1,
+    rowIdx: -1
+  },
+  isEditorEnabled: false,
+  firstEditorKeyPress: null
+};
+
+function connect(mapStateToProps, mapDispatchToProps, subscribeToEvents = () => {}) {
```

Comment:
@diogofcunha was trying to make this more generic, and now it feels like I'm just making a basic version of redux